src/main/java/br/com/openenade/api/universidade/UniversidadeService.java

```diff
@@ -13,68 +13,71 @@
-		}
-		universidade.setCampus(newCampus);
-		return this.repository.save(universidade);
+    public Universidade save(Universidade universidade) {
+        Optional<Municipio> optMunicipio =
+                this.municipioService.getOptionalByCodigo(universidade.getCampus().getCodigo());
+        Municipio newCampus;
+        if (optMunicipio.isPresent()) {
+            newCampus = optMunicipio.get();
+        } else {
+            newCampus = this.municipioService.save(universidade.getCampus());
+        }
+        universidade.setCampus(newCampus);
+        return this.repository.save(universidade);
 
-	}
+    }
 
-	public Collection<Universidade> getAll() {
-		return this.repository.findAll();
-	}
+    public Collection<Universidade> getAll() {
+        return this.repository.findAll();
+    }
 
-	// Issue #38
-	public Collection<Universidade> getAllByCodigoIES(Long codigoIES) {
+    // Issue #38
+    public Collection<Universidade> getAllByCodigoIES(Long codigoIES) {
 
-		return this.repository.findAllByCodigoIES(codigoIES);
+        return this.repository.findAllByCodigoIES(codigoIES);
 
-	}
+    }
 
-	public Universidade getUniversidadeByCodigoIES(Long codigoIES) {
-		Optional<Universidade> optUnivesidade = this.repository.findByCodigoIES(codigoIES);
-		if (optUnivesidade.isPresent()) {
-			return optUnivesidade.get();
-		} else {
-			throw new ResourceNotFound("" + codigoIES);
-		}
-	}
+    public Universidade getUniversidadeByCodigoIES(Long codigoIES) {
+        Optional<Universidade> optUnivesidade = this.repository.findByCodigoIES(codigoIES);
+        if (optUnivesidade.isPresent()) {
+            return optUnivesidade.get();
+        } else {
+            throw new ResourceNotFound("" + codigoIES);
+        }
+    }
 
-	public void deleteUniversidadeByCodigoIES(Long codigoIES) {
-		this.repository.deleteByCodigoIES(codigoIES);
-	}
+    public void deleteUniversidadeByCodigoIES(Long codigoIES) {
+        this.repository.deleteByCodigoIES(codigoIES);
+    }
 
-	public void deleteUniversidadesByMunicipioCodigo(Municipio campus) {
+    public void deleteUniversidadesByMunicipioCodigo(Municipio campus) {
 
-		this.repository.deleteAllByCampus(campus);
-	}
+        this.repository.deleteAllByCampus(campus);
+    }
 
-	public Optional<Universidade> getUniversidadeById(Long codigoIES, Long codigoMunicipio) {
+    public Optional<Universidade> getUniversidadeById(Long codigoIES, Long codigoMunicipio) {
 
-		Municipio municipio = this.municipioService.getMunicipioByCodigo(codigoMunicipio);
+        Optional<Municipio> municipio = this.municipioService.getOptionalByCodigo(codigoMunicipio);
 
-		UniversidadeId id = new UniversidadeId(codigoIES, municipio.getCodigo());
-		return this.repository.findById(id);
+        if (municipio.isPresent()) {
+            UniversidadeId id = new UniversidadeId(codigoIES, codigoMunicipio);
+            return this.repository.findById(id);
+        } else {
+            return Optional.ofNullable(null);
```

Comment:
I think that the return should be a ResourceNotFound that will be translated to a HttpStatus NotFound (404).

Reply:
Podemos pensar em algo melhor, porém, se Universidade não existir o contrato deste método diz que deve retornar um Optional onde isPresent() = false.

Reply:
Mas que fique claro, eu não acho que está perfeito a forma que eu fiz, vou pensar em algo melhor.

Reply:
Acredito que a forma que eu fiz agora deixou muito melhor.